src/qililab/digital/circuit_optimizer.py

```diff
@@ -162,23 +163,104 @@ def bunch_drag_gates(gate_list: list[gates.Gate]) -> list[gates.Gate]:
         Returns:
             list[gates.Gate]: list of gates of the transpiled circuit, with drag gates bunched."""
 
-        # Add bunching of Drag gates here:
-        # gate_list =
-        return gate_list
+        for idx1, drag1 in enumerate(gate_list):
+            # We are in the last gate, so there is no next one to merge with:
+            if idx1 == len(gate_list) - 1:
+                break
+
+            # Find a Drag gate to merge with:
+            if not isinstance(drag1, Drag):
+                continue
+
+            for idx2, drag2 in enumerate(gate_list[idx1 + 1 :]):
+                # Find next gate in the same qubit
+                if drag2 is None or drag2.qubits != drag1.qubits:
+                    continue
+
+                # If the next gate in the same qubit is not a Drag gate, we can't optimize:
+                if not isinstance(drag2, Drag):
+                    break
+
+                # If the next gate in the same qubit is a Drag gate, we can merge them:
+                new_drag: Drag | None = CircuitOptimizer.merge_consecutive_drags(drag1, drag2)
+
+                # If successful bunching, substitute old gates with new one:
+                if new_drag is not None:
+                    gate_list[idx1], drag1 = new_drag, new_drag
+                    gate_list[idx2 + idx1 + 1] = None
+
+                # If not successful bunching, we can't merge this gate, go next one:
+                if new_drag is None:
+                    break
```

Comment:
It's always more clear to have the breaking condition first, and otherwise continue with the execution flow.

```suggestion
                # If not successful bunching, we can't merge this gate, go next one:
                if new_drag is None:
                    break
                 
                gate_list[idx1], drag1 = new_drag, new_drag
                gate_list[idx2 + idx1 + 1] = None
```

Reply:
Nice improvement indeed, thanks!